GVRf/Framework/jni/engine/importer/assimp_importer.cpp

```diff
@@ -68,4 +76,178 @@ std::shared_ptr<Mesh> AssimpImporter::getMesh(int index) {
+{
+    for(int i=0; i < assimp_node->mNumMeshes; i++)
+    {
+        std::shared_ptr<SceneObject> gvr_scene_object(new SceneObject());
+
+        // Mesh
+        std::shared_ptr<Mesh> gvr_mesh = getMesh(assimp_node->mMeshes[i]);
+
+        // New render data object.
+        std::shared_ptr<RenderData> scene_object_render_data(new RenderData());
+
+        // Set the mesh to the render data.
+        scene_object_render_data->set_mesh(gvr_mesh);
+
+        // Set material.
+        aiMesh* assimp_mesh = assimp_scene->mMeshes[assimp_node->mMeshes[i]];
+        aiMaterial* assimp_material = assimp_scene->mMaterials[assimp_mesh->mMaterialIndex];
+
+        // Defines a shader with shader type 0. UNLIT_SHADER = 0
+        std::shared_ptr<Material> gvr_material(new Material(Material::ShaderType::UNLIT_SHADER));
+
+        // Actual texture image
+        aiString assimp_texture_file_name;
+        assimp_material->GetTexture(aiTextureType_DIFFUSE, i, &assimp_texture_file_name);
+        jstring texture_file_name = env->NewStringUTF(assimp_texture_file_name.C_Str());
+
+        // Default texture
+
+        bool repeat_flag = false;
+
+        if(gvr_mesh->getTextureRepeatFlag()) {
+            repeat_flag = true;
+        }
+
+        std::shared_ptr<Texture> default_texture(new BaseTexture(env, default_bitmap, repeat_flag));
+
+            if (method_ID == NULL) {
+                // No such method in Java side
+                // No interaction with Java side so apply default texture
+
+                gvr_material->setTexture("main_texture", default_texture);
+            } else {
+                if ((assimp_material->GetTextureCount(aiTextureType_DIFFUSE)) <= 0) {
+                    // No texture file found so apply default texture
+                    gvr_material->setTexture("main_texture", default_texture);
+                } else {
+                    // About to enter Java side.
+                    jobject actual_texture_bitmap = env->CallObjectMethod(gvr_context, method_ID, texture_file_name);
+                    if(actual_texture_bitmap == NULL)
+                    {
+                        // Null bitmap for texture file for mesh
+                        // Applying default texture
+                        gvr_material->setTexture("main_texture", default_texture);
+                    }
+                    else
+                    {
+                        // Back to native side from Java side
+                        // Applying actual texture
+                        std::shared_ptr<Texture> actual_texture(new BaseTexture(env, actual_texture_bitmap, repeat_flag));
+                        gvr_material->setTexture("main_texture", actual_texture);
+                    }
+                }
+            }
+
```

Comment:
Why there is space for this  if-else block. And also, I am a little bit confused about the use of default texture. It seems to me that if any error occurs while loading texture from resource, you will use the default texture instead. Should we rather quit the app with some error report? Just my personal opinion.